src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftBuilder.java

```diff
@@ -107,7 +107,7 @@ default void waitOnBuild(IClient client, long startTime, String bldId, TaskListe
 			if (Boolean.parseBoolean(getVerbose(overrides)))
 				listener.getLogger().println("\nOpenShiftBuilder bld state:  " + bldState);
 			
-			if (follow) {
+			if (follow && bldState.equals("Running") && stop == null) {
```

Comment:
Yes, I have rebuilt it and tried it out.
I added the stop == null as the catch to avoid the streaming being created multiple times.